app/src/main/java/com/github/onlynotesswent/ui/user/EditProfile.kt

```diff
@@ -121,3 +177,79 @@ fun ProfileScreen(navigationActions: NavigationActions, userViewModel: UserViewM
             }
       })
 }
+
+@SuppressLint("StateFlowValueCalledInComposition")
+@Composable
+fun ProfilePicture(
+    profilePictureTaker: ProfilePictureTaker,
+    userViewModel: UserViewModel,
+    profilePictureUri: MutableState<String>,
+    fileViewModel: FileViewModel,
+    isProfilePictureUpToDate: MutableState<Boolean>,
+    hasProfilePictureBeenChanged: MutableState<Boolean>,
+    localContext: Context
+) {
+
+  Box(modifier = Modifier.size(150.dp)) {
+    // Download the profile picture from Firebase Storage if it hasn't been downloaded yet
+    if (!isProfilePictureUpToDate.value &&
+        userViewModel.currentUser.value?.hasProfilePicture == true) {
+      fileViewModel.downloadFile(
+          userViewModel.currentUser.value!!.uid,
```

Comment:
`!!` throws `NullPointerException ` if `userViewModel.currentUser.value` is `null`